k8s upstream failure:
```10:49:49      k8s1-1.12: Status: Downloaded newer image for quay.io/cilium/cilium-builder:2018-11-01
10:49:49      k8s1-1.12:  ---> 80d50d3828dd
10:49:49      k8s1-1.12: Step 3/19 : LABEL maintainer="maintainer@cilium.io"
10:49:49      k8s1-1.12:  ---> Running in 4ef1f947e9e8
10:49:49      k8s1-1.12: Removing intermediate container 4ef1f947e9e8
10:49:49      k8s1-1.12:  ---> 259c032f3d90
10:49:49      k8s1-1.12: Step 4/19 : WORKDIR /go/src/github.com/cilium/cilium
10:49:49      k8s1-1.12:  ---> Running in b97740cf6606
10:49:49      k8s1-1.12: Removing intermediate container b97740cf6606
10:49:49      k8s1-1.12:  ---> f22a9cf6ee36
10:49:49      k8s1-1.12: Step 5/19 : COPY . ./
10:55:20      k8s1-1.12: failed to export image: failed to set parent sha256:f22a9cf6ee36d189e63af309f2cdda51b027a76f231603682b981988a4b62d81: unknown parent image ID sha256:f22a9cf6ee36d189e63af309f2cdda51b027a76f231603682b981988a4b62d81
10:55:20      k8s1-1.12: Makefile:144: recipe for target 'docker-image-no-clean' failed
10:55:20      k8s1-1.12: make: *** [docker-image-no-clean] Error 1
10:55:20  The SSH command responded with a non-zero exit status. Vagrant
10:55:20  assumes that this means the command failed. The output for this command
10:55:20  should be in the log above. Please read the output to determine what
10:55:20  went wrong.```

--- a/Documentation/gettingstarted/k8s-install-eks.rst
+++ b/Documentation/gettingstarted/k8s-install-eks.rst
@@ -82,6 +82,19 @@ leaving a node to be automatically be masqueraded.
 
        kubectl -n kube-system set env ds aws-node AWS_VPC_K8S_CNI_EXTERNALSNAT=true
        
+Prepare the nodes for Cilium
+============================
+
+Deploy the following DaemonSet to prepare all EKS nodes for Cilium:
+
+   .. code:: bash
+
+       kubectl -n kube-system apply -f \ |SCM_WEB|\/examples/kubernetes/node-init/eks-node-init.yaml
+
+This will mount the BPF filesystem and ensures that the filesystem is
+automatically mounted when the node is rebooted. Due to being a DaemonSet, any
+new node added to the cluster will automatically get initialized as well.
+
 .. include:: k8s-install-etcd-operator-steps.rst
 
 .. note::
@@ -89,6 +102,7 @@ leaving a node to be automatically be masqueraded.
    You may notice that the ``kube-dns-*`` pods get restarted. The
    ``cilium-operator`` will automatically restart CoreDNS if the pods are not
    managed by the Cilium CNI plugin.
+
 
 Validate the Installation
 =========================

--- a/examples/kubernetes/node-init/eks-node-init.yaml
+++ b/examples/kubernetes/node-init/eks-node-init.yaml
@@ -1,0 +1,72 @@
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: cilium-node-init
+  labels:
+    app: cilium-node-init
+spec:
+  template:
+    metadata:
+      labels:
+        app: cilium-node-init
+    spec:
+      tolerations:
+      - operator: Exists
+      hostPID: true
+      hostNetwork: true
+      containers:
+        - name: node-init
+          image: gcr.io/google-containers/startup-script:v1
+          imagePullPolicy: IfNotPresent
+          securityContext:
+            privileged: true
+          env:
+          - name: STARTUP_SCRIPT
+            value: |
+              #!/bin/bash
+
+              set -o errexit
+              set -o pipefail
+              set -o nounset
+
+              if [[ ! -f /tmp/cilium-installed-v1 ]]; then
+                echo "Installing BPF filesystem mount"
+
+                cat >/tmp/sys-fs-bpf.mount <<EOF
+              [Unit]
+              Description=Mount BPF filesystem (Cilium)
+              Documentation=http://docs.cilium.io/
+              DefaultDependencies=no
+              Before=local-fs.target umount.target
+              After=swap.target
+
+              [Mount]
+              What=bpffs
+              Where=/sys/fs/bpf
+              Type=bpf
+
+              [Install]
+              WantedBy=multi-user.target
+              EOF
+
+                if [ -d "/etc/systemd/system/" ]; then
+                  mv /tmp/sys-fs-bpf.mount /etc/systemd/system/
+                  echo "Installed sys-fs-bpf.mount to /etc/systemd/system/"
+                elif [ -d "/lib/systemd/system/" ]; then
+                  mv /tmp/sys-fs-bpf.mount /lib/systemd/system/
+                  echo "Installed sys-fs-bpf.mount to /lib/systemd/system/"
+                fi
+
+                systemctl enable sys-fs-bpf.mount
+                systemctl start sys-fs-bpf.mount
+
+                echo "Link information:"
+                ip link
+
+                echo "Routing table:"
+                ip route
+
+                echo "Node initialization complete"
+
+                touch /tmp/cilium-installed-v1
+              fi

--- a/pkg/kvstore/allocator/allocator.go
+++ b/pkg/kvstore/allocator/allocator.go
@@ -406,6 +406,11 @@ func (a *Allocator) createValueNodeKey(key string, newID idpool.ID) error {
 		return fmt.Errorf("unable to create value-node key '%s': %s", valueKey, err)
 	}
 
+	// mark the key as verified in the local cache
+	if err := a.localKeys.verify(key); err != nil {
+		log.WithError(err).Error("BUG: Unable to verify local key")
+	}
+
 	return nil
 }
 
@@ -452,11 +457,6 @@ func (a *Allocator) lockedAllocate(ctx context.Context, key AllocatorKey) (idpoo
 		if err = a.createValueNodeKey(k, value); err != nil {
 			a.localKeys.release(k)
 			return 0, false, fmt.Errorf("unable to create slave key '%s': %s", k, err)
-		}
-
-		// mark the key as verified in the local cache
-		if err := a.localKeys.verify(k); err != nil {
-			log.WithError(err).Error("BUG: Unable to verify local key")
 		}
 
 		return value, false, nil

--- a/test/helpers/kubectl.go
+++ b/test/helpers/kubectl.go
@@ -1125,7 +1125,9 @@ func (kub *Kubectl) CiliumExecContext(ctx context.Context, pod string, cmd strin
 // CiliumExec runs cmd in the specified Cilium pod.
 // Deprecated: use CiliumExecContext instead
 func (kub *Kubectl) CiliumExec(pod string, cmd string) *CmdRes {
-	return kub.CiliumExecContext(context.Background(), pod, cmd)
+	ctx, cancel := context.WithTimeout(context.Background(), HelperTimeout)
+	defer cancel()
+	return kub.CiliumExecContext(ctx, pod, cmd)
 }
 
 // CiliumExecUntilMatch executes the specified command repeatedly for the

